server/scores.js

```diff
@@ -4,27 +4,80 @@ const express = require('express')
 const Promise = require('bluebird')
 const { MongoClient, ObjectID } = require('mongodb')
 const { authroizationMiddlware } = require('@first-lego-league/ms-auth')
+const Configuration = require('@first-lego-league/ms-configuration')
 
 const DEFAULTS = require('./defaults')
 
 const mongoUrl = process.env.MONGO_URI || DEFAULTS.MONGO
 
 const router = express.Router()
 
+const STATUS = {
+  GOOD: 'ok',
+  CONFIGURATION_ERROR: 'config-error',
+  LOUD_FAIL: 'loud-fail'
+}
+
+const ERROR = {
+  TEAM_NUMBER: 'team number ',
+  SCORE: 'score ',
+  MATCH: 'match ',
+  NONE: ''
+}
+
 const connectionPromise = MongoClient
   .connect(mongoUrl, { promiseLibrary: Promise, useNewUrlParser: true })
   .then(client => client.db().collection('scores'))
 
+function _validateScore (score) {
+  let retError = { 'status': STATUS.GOOD, 'errors': ERROR.NONE }
+
+  Configuration.get('autoPublish').then(autoPublishSetting => {
+    const publishBool = autoPublishSetting
+    return publishBool
+  }).then(result => {
+    score.published = result
+  }).catch(() => {
+    retError.status = STATUS.CONFIGURATION_ERROR
+    return retError
+  })
+
+  if (typeof score.teamNumber !== 'number') { retError.errors += ERROR.TEAM_NUMBER }
+  if (score.score == null) { retError.errors += ERROR.SCORE }
+  if (score.match == null) { retError.errors += ERROR.MATCH }
+
+  if (retError.errors !== ERROR.NONE) { retError.status = STATUS.LOUD_FAIL }
+
+  return retError
+}
+
 const adminAction = authroizationMiddlware(['admin', 'scorekeeper', 'development'])
 
 router.post('/create', (req, res) => {
-  connectionPromise
-    .then(scoringCollection => scoringCollection.save(req.body))
-    .then(() => res.status(201).send())
-    .catch(err => {
-      req.logger.error(err.message)
-      res.status(500).send('A problem occoured while trying to save score.')
-    })
+  const scoreValidation = _validateScore(req.body)
+
+  switch (scoreValidation.status) {
+    case STATUS.GOOD:
+      connectionPromise
+        .then(scoringCollection => {
+          scoringCollection.save(req.body)
+        })
+        .then(() => {
+          res.status(201).send()
+        })
+        .catch(err => {
+          req.logger.error(err.message)
+          res.status(500).send('A problem occoured while trying to save score.')
+        })
+      break
+    case STATUS.LOUD_FAIL:
+      req.logger.error('Invalid score, missing ' + scoreValidation.errors + '. ' + scoreValidation.status + '.')
+      res.status(422).send('Invalid score, missing ' + scoreValidation.errors)
+      break
+    case STATUS.CONFIGURATION_ERROR:
+      res.status(500).send('Could not load configuration')

```

Comment:
A configuration error did be a generic 500 like in line 70. It doesn't need a special case